server/src/test/java/org/elasticsearch/index/mapper/NumberFieldMapperTests.java

```diff
@@ -248,6 +249,28 @@ public void testOutOfRangeValues() throws IOException {
         }
     }
 
+    public void testDimension() throws IOException {
+        MapperService mapperService = createMapperService(fieldMapping(b -> {
+            minimalMapping(b);
+            b.field("dimension", false);
+        }));
+
+        MappedFieldType fieldType = mapperService.fieldType("field");
+        assertThat(fieldType, instanceOf(NumberFieldMapper.NumberFieldType.class));
+        NumberFieldMapper.NumberFieldType ft = (NumberFieldMapper.NumberFieldType) fieldType;
+        assertFalse(ft.isDimension());
+
+        try {
+            createMapperService(fieldMapping(b -> {
+                minimalMapping(b);
+                b.field("dimension", true);
+            }));
+            fail("Mapper parsing exception expected for non-integer numeric with dimension parameter set");
```

Comment:
I *think* `expectThrows` would be a little more readable here.

Reply:
Right, I merely copied the code in the method above. I fixed both